exec/exploratory_plots.R

```diff
@@ -179,15 +185,65 @@ if (is.null(opt$final_assay)){
   }
 }
 
-# Guess which assays aren't logged and log them. Don't like this, but need a
-# cleaner solution
+if (is.null(opt$log2_assays)) {
+
+  print("--log2_assays param not set, will guess which input assays need to be log2.")
+  unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+  for (unlogged_expression_type in names(unlogged)[unlogged]) {
+    if (unlogged_expression_type %in% names(assay_data)) {
+      assay_data[[unlogged_expression_type]] <- log2(assay_data[[unlogged_expression_type]])
+    }
+  }
 
-unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+} else if (opt$log2_assays == "") {
+
+  print("--log2_assays param set to empty string, will not apply log2.")
+
+} else {
+  if (is_valid_positive_integer_vector(opt$log2_assays)) {
+
+    print("--log2_assays param set to list of int, will apply log2 to specified assays.")
+    unlogged <- as.integer(simpleSplit(opt$log2_assays))
```

Comment:
```suggestion
    unlogged <- unique(as.integer(simpleSplit(opt$log2_assays)))
    
    invalid_assays <- unlogged[! unlogged %in% 1:length(assay_data)]
    if (length(invalid_assays) > 0){
        stop(paste0("Invalid assays: ", paste(invalid_assays, collapse=', ')))
    }
```

This will suffice for checks I think - you can delete the below. We'll just uniquify to make sure we don't log things twice.